roles/kubernetes/node/tasks/kubelet.yml

```diff
@@ -17,7 +17,7 @@
   template:
     src: "kubelet.env.{{ kubeletConfig_api_version }}.j2"
     dest: "{{ kube_config_dir }}/kubelet.env"
-    setype: "{{ (preinstall_selinux_state == 'enforcing') | ternary('t_etc', omit) }}"
+    setype: "{{ (preinstall_selinux_state == 'enforcing') | ternary('etc_t', omit) }}"
```

Comment:
You should also set the correct type in permissive mode

Reply:
Good idea, I changed the condition to `!= 'disabled'`.

Reply:
This is now needed to fix current deployments, but I'm wondering why it was introduced in the first place

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1957840 may shed some light.

`systemd` runs as `init_t` domain and it is not allowed to read `kubernetes_files_t` which was recently introduced in Fedora 33/RHEL8.4. The fix was to allow `init_t` access to `kubernetes_files_t` but apparently the fix is still not working properly. Because of this we proactively change the context of `/etc/kubernetes/kubelet.env` to `etc_t` to allow systemd to read it and apply the environment to the kubelet service.

Reply:
/cc @spaced  since you authored the initial change can you help with some extra context?